pkg/api/osa/vlabs/types.go

```diff
@@ -0,0 +1,99 @@
+package vlabs
+
+// OpenShiftCluster complies with the ARM model of resource definition in a JSON
+// template.
+type OpenShiftCluster struct {
+	ID         string                `json:"id,omitempty"`
+	Location   string                `json:"location,omitempty"`
+	Name       string                `json:"name,omitempty"`
+	Plan       *ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags       map[string]string     `json:"tags,omitempty"`
+	Type       string                `json:"type,omitempty"`
+	Properties Properties            `json:"properties,omitempty"`
+}
+
+// ResourcePurchasePlan defines the resource plan as required by ARM for billing
+// purposes.
+type ResourcePurchasePlan struct {
+	Name          string `json:"name,omitempty"`
+	Product       string `json:"product,omitempty"`
+	PromotionCode string `json:"promotionCode,omitempty"`
+	Publisher     string `json:"publisher,omitempty"`
+}
+
+// Properties represents the cluster definition.
+type Properties struct {
+	ProvisioningState ProvisioningState `json:"provisioningState,omitempty"`
+	OpenShiftVersion  string            `json:"openShiftVersion,omitempty"`
+	// TODO: need to clarify external API for PublicHostname and decide what to
+	// implement when.  Allow users to specify a PublicHostname then have them
+	// create a CNAME to a FQDN we return?  Allow users to not specify and we
+	// return a FQDN?  In which case, how will the plugin know the FQDN?
```

Comment:
I might not have full picture of the use case but what is the plugin being referred to here?

Reply:
the OpenShift plugin in azure-helm.  It uses the publicHostName in a couple places.  Notably, generating certificates, kube config, and various api object configs